lib/api/src/externals/function.rs

```diff
@@ -562,14 +574,70 @@ impl Function {
 
         match &self.definition {
             FunctionDefinition::Wasm(wasm) => {
-                self.call_wasm(&wasm, params, &mut results)?;
+                self.call_wasm(&wasm, params, &mut results, true)?;
             }
             _ => unimplemented!("The function definition isn't supported for the moment"),
         }
 
         Ok(results.into_boxed_slice())
     }
 
+    /// Call the `Function` function, without catching traps.
+    ///
+    /// This is a performance optimization for making function calls in batch.
+    /// It must be called within a callback passed to `wasmer::vm::catch_traps`.
+    ///
+    /// Depending on where the Function is defined, it will call it.
+    /// 1. If the function is defined inside a WebAssembly, it will call the trampoline
+    ///    for the function signature.
+    /// 2. If the function is defined in the host (in a native way), it will
+    ///    call the trampoline.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # use wasmer::{imports, wat2wasm, Function, Instance, Module, Store, Type, Value, vm};
+    /// # let store = Store::default();
+    /// # let wasm_bytes = wat2wasm(r#"
+    /// # (module
+    /// #   (func (export "sum") (param $x i32) (param $y i32) (result i32)
+    /// #     local.get $x
+    /// #     local.get $y
+    /// #     i32.add
+    /// #   ))
+    /// # "#.as_bytes()).unwrap();
+    /// # let module = Module::new(&store, wasm_bytes).unwrap();
+    /// # let import_object = imports! {};
+    /// # let instance = Instance::new(&module, &import_object).unwrap();
+    /// #
+    /// let sum = instance.exports.get_function("sum").unwrap();
+    /// let vmctx = sum.get_vmctx();
+    /// unsafe {
+    ///     vm::catch_traps(vmctx, || {
+    ///         for i in 0..10 {
+    ///             assert_eq!(sum.call(&[Value::I32(1), Value::I32(2)]).unwrap().to_vec(), vec![Value::I32(3)]);
+    ///         }
+    ///     });
+    /// }
+    /// ```
+    pub fn call_unchecked(&self, params: &[Val]) -> Result<Box<[Val]>, RuntimeError> {
```

Comment:
This function must be `unsafe` and it should have all prerequisites to call it correctly documented in a `# Safety` section.